code/game/objects/items/weapons/implants/exile2.dm

```diff
@@ -1,39 +1,125 @@
 
-
 /obj/item/weapon/implant/exile
 	name = "exile"
-	desc = "Prevents you from returning from the asteroid"
+	desc = "Prevents returning to where you were implanted"
+	var/illegalZ = null
+	var/mob/living/theExile = null
+	var/siteOfImplant = null
+	var/beingDeported = FALSE
+	var/beenSpaced = FALSE
+	var/disablePhrase = ""
+	var/list/zlevels = list(STATION_Z, TELECOMM_Z, DERELICT_Z, ASTEROID_Z, SPACEPIRATE_Z)
+	var/zBanishment_key
 
 /obj/item/weapon/implant/exile/get_data()
 	var/dat = {"
 <b>Implant Specifications:</b><BR>
 <b>Name:</b> Nanotrasen Employee Exile Implant<BR>
-<b>Implant Details:</b> The host of this implant will be prevented from returning to the station."}
+<b>Implant Details:</b> The host of this implant will be prevented from returning to the implant location.<BR>
+<b>For non-permanent use a disable phrase may be assigned on application.<BR>"}
 	return dat
 
-/obj/item/weapon/implant/exile/implanted(mob/M)
-	if(!istype(M, /mob/living/carbon))
-		return 0
-	var/mob/living/carbon/I = M
-	to_chat(I, "<span class='notice'>You shiver as you feel a weak bluespace void surround you.</span>")
-	I.locked_to_z = ASTEROID_Z
+/obj/item/weapon/implant/exile/implanted(mob/source as mob)
+	theExile = source
+	disablePhrase = input("Choose phrase to disable:") as text
+	var/list/replacechars = list("'" = "", "\"" = "", ">" = "", "<" = "", "(" = "", ")" = "")
+	disablePhrase = sanitize_simple(disablePhrase, replacechars)
+	addHear()
+	illegalZ = theExile.z
+	siteOfImplant = get_turf(theExile)
+	zlevels -= illegalZ
+	to_chat(theExile, "<span class='notice'>You shiver as you feel a weak, unsettling film surround you.</span>")
+	zBanishment_key = theExile.on_moved.Add(src, "zBan")
 	return 1
 
+/obj/item/weapon/implant/exile/proc/zBan()
+	var/turf/T = get_turf(src)
+	switch(beenSpaced)
+		if(FALSE)
+			if(T.z != illegalZ)
+				beenSpaced = TRUE
+		if(TRUE)
+			if((T.z == illegalZ) && (!beingDeported))
+				beingDeported = TRUE
+				teleDeport()
+
+/obj/item/weapon/implant/exile/proc/teleDeport()
+	to_chat(theExile, "<span class='notice'>Your insides churn and your skin tingles. Your [part] is emitting a low hum.</span>")
+	spawn(10 SECONDS)
+		var/turf/T = get_turf(src)
+		if(T.z == illegalZ)
+			var/warpZ = pick(zlevels)
+			var/warpTo = locate(rand(5,world.maxx - 10), rand(5, world.maxy - 10), warpZ)
+			var/W = get_turf(warpTo)
+			if(!istype(W, /turf/space))
+				to_chat(theExile, "<span class='notice'>Your [part] buzzes. The tingling stops.</span>")
+				sleep(3 SECONDS)
+				beingDeported = FALSE
+				return
+			do_teleport(theExile, warpTo, 1)
+			theExile.Knockdown(3)
+			theExile.Stun(3)
+			theExile.adjustBruteLoss(rand(0,5))
+			theExile.adjustCloneLoss(rand(0,5)) //Uh oh it missed a few chromosomes
+		else
+			to_chat(theExile, "<span class='notice'>Your [part] emits a feint chime. The tingling stops.</span>")
+		beingDeported = FALSE
+
+/obj/item/weapon/implant/exile/emp_act()
+	if(malfunction)
+		return
+	malfunction = 1
```

Comment:
pick one:
- consistently set this to fixed values like here
- consistently decrease/increase this value, like you did in the spawn()